pyscf/lib/agf2/CMakeLists.txt

```diff
@@ -17,6 +17,6 @@ add_library(agf2 SHARED
 add_dependencies(agf2 cvhf)
 
 set_target_properties(agf2 PROPERTIES
-  LIBRARY_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR})
+  LIBRARY_OUTPUT_DIRECTORY ${PYSCF_LIBOUTDIR_ROOT})
```

Comment:
This change will break the search paths  in load_library function. Please do not change any of the cmake configurations.

Reply:
I really did try not to break anything, I swear! See [these lines](https://github.com/chillenb/pyscf/blob/bc6dab5ed7ed0c2d9bd909603a371f52e14d42e3/pyscf/lib/CMakeLists.txt#L22C1-L26C8):

```
if(DEFINED SKBUILD)
  set(PYSCF_LIBOUTDIR_ROOT ${PROJECT_BINARY_DIR})
else()
  set(PYSCF_LIBOUTDIR_ROOT ${PROJECT_SOURCE_DIR})
endif()
```

on the other hand, I get it. "if it ain't broke, don't fix it."